pysmt/oracles.py

```diff
@@ -168,10 +169,73 @@ def get_theory(self, formula):
         """Returns the thoery for the formula."""
         return self.walk(formula)
 
-
 # EOC TheoryOracle
 
 
+# Operators for which Args is an FNode
+DEPENDENCIES_SIMPLE_ARGS = (set(op.ALL_TYPES) - \
+                           (set([op.SYMBOL, op.FUNCTION]) | op.QUANTIFIERS | op.CONSTANTS))
+
+class FreeVarsOracle(pysmt.walkers.DagWalker):
```

Comment:
Lets keep this for a future. I think this goes in the direction of having an implementation in which we do Free Variables vs All Variables. Lets open a ticket with all "versions" that we want to have.